artifact-version-diff/src/main/java/io/quarkus/qe/GenerateReport.java

```diff
@@ -0,0 +1,341 @@
+package io.quarkus.qe;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+public class GenerateReport {
+
+    public static final String HTML_BASE_START = """
+            <!DOCTYPE html>
+            <html>
+            <style>
+                table, td, th {
+                    border: 1px solid;
+                    vertical-align: top;
+                    text-align: left;
+                    padding: 5px;
+                }
+
+                table {
+                    width: 100%;
+                    border-collapse: collapse;
+                }
+                .newer {
+                    background-color: #FFDD88;
+                }
+                .older {
+                    background-color: #EE9999;
+                }
+            </style>
+            <body>
+            <div>
+                <table>""";
+
+    public static final String HTML_BASE_END = """
+                </table>
+            </div>
+            </body>
+            </html>""";
+    public static final String ALLOWED_ARTIFACTS_BASE = "_allowed_artifacts.yaml";
+    public TreeMap<String, Artifact> differentArtifacts = new TreeMap<>();
+
+    Map<String, List<String>> allowedArtifacts;
+
+    private boolean majorMinorVersionDiffer = false;
+    private boolean patchOrOthersVersionDiffer = false;
+
+    public GenerateReport() {
+        createAllowedHashMap();
+    }
+
+    public void generateReport() {
+        String basePath = Objects.requireNonNull(System.getProperty("path-to-quarkus-repository"));
+        String versionDiffOutputFileName = Objects.requireNonNull(System.getProperty("version-plugin.report-output-file"));
+        List<Path> dependencyDiffFiles = generateListOfDiffFiles();
+
+        for (Path path : dependencyDiffFiles) {
+
+            try (Stream<String> lines = Files.lines(path)) {
+                lines.forEach(line -> {
+                    Pattern artifactPattern = Pattern.compile("[\\w.-]+:[\\w.-]+");
+                    Pattern versionsPattern = Pattern.compile("[\\S\\d.]+.->.+");
```

Comment:
Looking at this I see I didn't add any comment.

I'm using this to parse the file which was generated by version plugin. The file look like this:
```
The following differences were found:

  com.github.javaparser:javaparser-core . 3.25.8 -> 3.25.8.redhat-00001
  com.aayushatharva.brotli4j:brotli4j ... 1.16.0 -> 1.16.0.redhat-00003
.....
  io.smallrye.stork:stork-microprofile-config  2.5.0 -> 2.5.0.redhat-00001
  io.smallrye.stork:stork-test-utils ...... 2.5.0 -> 2.5.0.redhat-00001

The following property differences were found:

  none
```

So with using these regex I get artifact (` io.smallrye.stork:stork-test-utils`) and versions (`2.5.0 -> 2.5.0.redhat-00001`) where I only use artifact name as key in map to get versions (expecting there can be more then one version) 

>Also - maybe this would be more suitable as a function of Artifact rather than having it in the logic of generation of the webpage?

Thinking about it in this I use `Artifact` class just as "database" used to store version and modules (it's path) where these version will be used. The name of artifact is stored as key in map. + This is part for preparing data for report so I would leave it i this part.

Reply:
Updated the method and add coments